src/WalkingModule/src/Module.cpp

```diff
@@ -579,13 +594,30 @@ bool WalkingModule::updateModule()
             }
         }
 
+        // check if the stance phase is started
+        double treshold = 0.01;
+        if(m_isStancePhaseStarting &&
+           iDynTree::toEigen(m_DCMVelocityDesired.front()).norm() < treshold)
+        {
+            m_stancePhaseCounter--;
+            if(m_stancePhaseCounter == 0)
+            {
+                m_isStancePhase = true;
+                m_isStancePhaseStarting = false;
+            }
+        }
+
+        // TODO remove me
+        yInfo() << "m_isStancePhase " << m_isStancePhase << " isStancePhaseStarting " << m_isStancePhaseStarting << " m_stancePhaseCounter " << m_stancePhaseCounter;
```

Comment:
Can't we have this info from the planner?

Reply:
Moved in https://github.com/robotology/walking-controllers/pull/41/commits/9760be0556d233e940c1faba357e1d48f3f54cf8